REVIEWING.md

```diff
@@ -0,0 +1,125 @@
+# Reviewing PRs at TensorZero
+
+We require a code review of every PR to the repository.
+Code review itself is an important part of the engineering process and should allow us to catch issues with correctness, developer experience, and conformance to a spec.
+In this document, we list some guidelines (a checklist) to follow when reviewing PRs so that we do not forget to check the necessary boxes.
+We omit changes that will be automatically caught by CI.
+
+## Reviewer checklist
+
+### General
+
+* Does the PR have a clear purpose? Could it be split into more focused changes?
+* Does the PR add any dependencies? Is there a good justification for adding them? Are we sure we want to depend on them?
+
```

Comment:
maybe a note about code quality - is there any unnecessary complexity? is the codebase in a better state than before the PR?